### ISSUE TITLE
support equivariant neural networks

### ISSUE THREAD

**Issue by CarloLucibello** [CarloLucibello/GraphNeuralNetworks.jl#221]
- [ ] General forms using spherical harmonics
https://www.nature.com/articles/s41467-022-29939-5
https://docs.e3nn.org/en/latest/guide/convolution.html

- [x] Simpler equivariant layers
https://proceedings.mlr.press/v139/satorras21a.html
https://github.com/lucidrains/egnn-pytorch
https://docs.dgl.ai/en/0.9.x/generated/dgl.nn.pytorch.conv.EGNNConv.html

**Comment by Dsantra92** [comment]
I was working with e3nn and nequip this summer and would love to see it in Julia. Something like e3nn.jl sounds exciting.

**Comment by CarloLucibello** [comment]
have you compared them to the simpler EGNN for your tasks?

**Comment by CarloLucibello** [comment]
Test accuracy. Implementation wise they are very different, e3nn much harder to implement

**Comment by Dsantra92** [comment]
No, not yet.

**Comment by Dsantra92** [comment]
Started writing this [e3nn.jl](https://github.com/Dsantra92/e3nn.jl) library.
